plutus-pab/src/Plutus/PAB/Core/ContractInstance/BlockchainEnv.hs

```diff
@@ -48,23 +54,23 @@ import System.Random
 -- | Connect to the node and write node updates to the blockchain
 --   env.
 startNodeClient ::
-     FilePath -- ^ Socket to connect to node
-  -> NodeMode -- ^ Whether to connect to real node or mock node
-  -> Maybe Int  -- ^ How much history do we remember for rollbacks
-  -> SlotConfig -- ^ Slot config used by the node
-  -> NetworkId -- ^ Cardano network ID
-  -> Point
+     Config -- ^ PAB's config
+  -> ProtocolParameters -- ^ Node's protocol parameters
   -> InstancesState -- ^ In-memory state of running contract instances
   -> IO BlockchainEnv
-startNodeClient socket mode rollbackHistory slotConfig networkId resumePoint instancesState = do
-    env <- STM.atomically $ emptyBlockchainEnv rollbackHistory slotConfig
-    case mode of
+startNodeClient config params instancesState = do
+    let Config { nodeServerConfig = PABServerConfig{pscSocketPath, pscSlotConfig, pscNodeMode, pscNetworkId = NetworkIdWrapper networkId}
+               , developmentOptions = DevelopmentOptions{pabRollbackHistory, pabResumeFrom} } = config
+    let ProtocolParameters{protocolParamUTxOCostPerWord = mCostPerWord } = params
+    let coinsPerUTxOWord = fromMaybe (fromJust $ protocolParamUTxOCostPerWord def) mCostPerWord
+    env <- STM.atomically $ emptyBlockchainEnv pabRollbackHistory pscSlotConfig coinsPerUTxOWord
```

Comment:
I decided to pass `Config` and `ProtocolParameters` directly. `startNodeClient` depends on too many params to use arguments here to my taste.